docs/design/storage-node-windows-installer.md

```diff
@@ -13,73 +13,64 @@ Similarly, Docker ends up using more resources than running natively on Windows.
-
-### Service
-
-We need to Storage Node to implement Windows service API, as shown in:
-
-* Modify storage node/automatic updater startup code to implement [ServiceMain](https://docs.microsoft.com/en-us/windows/win32/api/winsvc/nc-winsvc-lpservice_main_functiona).
-   * See [golang/sys](https://github.com/golang/sys/blob/master/windows/svc/example/service.go)
-* Create inbound firewall rule using the following Powershell command: `New-NetFirewallRule -DisplayName "Storj v3" -Direction Inbound –Protocol TCP –LocalPort 28967 -Action allow`
-
-This means that Windows handles starting and restarting the binary in the background.
+The high-level idea is to create an installer using the [WiX toolset](https://wixtoolset.org) and make Storage Node a Windows service.
+
+The installer shall:
+* Display a GUI for collecting user configuration about:
+  * Wallet address
+  * Email
+  * External address/port
+  * Advertised bandwidth 
+  * Advertised Storge
+  * Identity directory
+  * Installation directory
+  * Storage directory
+* Install the storage node binary in the installation directory.
+* Generate `config.yaml` file with the user configuration.
+* Register the storage node binary as a Windows Service: https://wixtoolset.org/documentation/manual/v3/xsd/util/serviceconfig.html
+* Create a firewall exception for port 28967: https://wixtoolset.org/documentation/manual/v3/xsd/firewall/firewallexception.html
+* Create shortcut for opening the Dashboard
+* Add installation directory to user PATH (optional)
+* Open the dashboard when the installation is complete (optional)
+* Install the auto-updater binary and run it as a Windows service (if auto-updater is implemented at this point)
+
+We must ensure that both the storage node binary and the MSI installer are signed with Storj code-sign certificate to avoid warning popups to users.
+
+## Rationale
+
+The initial idea was to use [go-msi](https://github.com/mh-cbon/go-msi) for creating the Windows installer. It utilizes the WiX toolset to build the actual installer.
+
+There are a number of reasons to use the WiX toolset directly instead of go-msi:
+* More flexibility as we can access all WiX toolset features directly instead of through the go-msi wrapper
+* One less dependency
+* The go-msi project is stall for 2 years. None of the issues and question has been answered for the last year.
+* The go-msi project lacks support for Windows Services and Firewall Rules, although the WiX toolset natively supports them.
 
 ## Implementation
 
-1) Modify the storage node startup to implement ServiceMain so that the binary is considered a Windows Service.
-2) Create script for registering binary as a Windows Background Service. (sc.exc command)
-3) Create wix.json file for building an MSI.
-4) Update the build process to include the creation of the MSI. 
-5) Ensure the windows installer is working properly
-  * Install Storage Node using msi.
+1. Implement ServiceMain in the storage node binary.
```

Comment:
@egonelbre do you remember why we need to implement `ServiceMain`? Is it for log rotation?

Reply:
This is how services handle requests from Windows. From example in Service Manager you can select "Restart Service"... the "service main" thing would handle those commands.

See example service here https://github.com/golang/sys/tree/master/windows/svc/example.

Reply:
OK... But what shall we implement for ServiceMain?

If we don't do anything, I assume the storage node binary will start as usual on "Start Service" and "Restart Service" events, and it will receive an ordinary terminate process signal on the "Stop Service" event.

Reply:
"stop" mainly I guess, to ensure that we close databases and things during computer restart... I'm not sure what will happen if we don't implement it.

Reply:
What does happen if I hit Ctrl+C in the terminal of the storage node process? Does it shutdown gracefully?

Reply:
I think it should, however I haven't tested it.

Reply:
OK. I will move this to the Open Issues section. We should test and confirm via debug logs that shutdown is graceful during "Stop/Restart Service". If not, we should implement ServiceMain,